Skill Calculator calculates XP on focusout

zacharydwaller opened the pull request:
Closes #10387 

Currently in order to input a number for level or experience a user must press enter in order for the input to be registered. This change makes it so that the user only needs to unfocus the text input for it to calculate the XP.

This improves usability for users that prefer to use tab over Enter, and in cases where XP/Level values are pasted.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java:
```diff
@@ -413,4 +423,44 @@ private boolean slotContainsText(UIActionSlot slot, String text)
+			@Override
+			public void focusLost(FocusEvent e)
+			{
+				try
```
This feels very hacky. I think it would make more sense to have one listener per field and make this just callback to a Consumer/Callable. That way it could work similar to the action listeners- `field.addFocusListener(buildListener(e ->onFieldTargetXPUpdated()))`.  Also if you only want to override one method you can use `FocusAdapter`.

zacharydwaller commented:
Back from vacation - I replaced the FocusListener with a FocusAdapter so I wouldn't have to override the focusGained. I also removed all the if/elses, the FocusAdapter is just passed a Consumer which is calls on focusLost. This lets adding a focus listener have the same usage as adding any other listener.
